pymeasure/instruments/lakeshore/lakeshore421.py

```diff
@@ -54,20 +54,29 @@ def __init__(self, resource_name, **kwargs):
         super(LakeShore421, self).__init__(
             resource_name,
             "Lake Shore 421 Gaussmeter",
-            read_termination='\r',
-            write_termination='\n',
             command_delay=0.05,
             **kwargs
         )
 
-        self.adapter.connection.set_visa_attribute(
-            visa_const.VI_ATTR_ASRL_BAUD, 9600)
-        self.adapter.connection.set_visa_attribute(
-            visa_const.VI_ATTR_ASRL_DATA_BITS, 7)
-        self.adapter.connection.set_visa_attribute(
-            visa_const.VI_ATTR_ASRL_STOP_BITS, visa_const.VI_ASRL_STOP_ONE)
-        self.adapter.connection.set_visa_attribute(
-            visa_const.VI_ATTR_ASRL_PARITY, visa_const.VI_ASRL_PAR_ODD)
+        if isinstance(self.adapter, VISAAdapter):
+            self.adapter.connection.baud_rate = 9600
+            self.adapter.connection.data_bits = 7
+            self.adapter.connection.stop_bits = 10
+            self.adapter.connection.parity = 1
+
+            self.adapter.connection.read_termination = '\r'
+            self.adapter.connection.write_termination = '\n'
+
+        elif isinstance(self.adapter, SerialAdapter):
+            self.adapter.connection.baudrate = 9600
+            self.adapter.connection.bytesize = 7
+            self.adapter.connection.stopbits = 1
+            self.adapter.connection.parity = "O"
+            self.adapter.connection.timeout = 1
+
+            # Ensure a \n is written after every command
+            write = self.adapter.write
+            self.adapter.write = lambda command: write(command + '\n')
```

Comment:
Huh, I was not aware that the format for the arguments (parity, stop_bits) differs between pyvisa and serial!
We should think if we should reduce priority/support for SerialAdapter and long-term replace it with VISAAdapter. 

I think in this case, we can only hit the elif clause if people passed a finished SerialAdapter (instead of naming the port with a string). In this special case, I think it's safe to assume that people can configure/have configured the adapter correctly themselves (otherwise, why pass an adapter instance?), so I would drop the elif clause.
It's probably harder to argue in case of the patched read_termination, but placing this here is imo really ugly, we should rather fix #160 instead.

Reply:
Argh this is so messy sometimes. :-/

Reply:
Yes, I agree it is indeed a bit ugly.
You're argument about people passing a SerialAdapter is sound I think, so I can drop the else clause.